### DISCUSSION TITLE
Help Mirth X Mumps

### DISCUSSION THREAD

**Discussion by rojao123** [nextgenhealthcare/connect#5163] [General]
Hello, I have an integration task between our RIS medical system and the worklist, our system is developed in MUMPS, an old and unknown language. It is possible to export to XML files, however, I don't know how to read the XML files in Mirth and send this data in a format understandable by the Worklist service. Could someone help me please?

**Comment by tonygermano** [comment]
Reading XML files are a simple and common operation in mirth. Use a File Reader to read a file. Set the data type to XML if that's what it is. If you don't know how to set up those two things, I would recommend taking a look at the User Guide.

I don't think anyone can help you with the second part unless you first describe what a "format understandable by the Worklist service" looks like.

**Reply by rojao123** [reply]
Good afternoon, I have an XML file that is generated by my system and I need to read this XML file to include it in my database. Below is an example of the generated XML file. I need to throw this information into my microsoft sql database.

<?xml version='1.0' encoding='ISO-8859-1' standalone='no'?>
<MWL_ITEM>
<SITE>HNSC</SITE>
<PATIENT_ID>45993998</PATIENT_ID>
<PATIENT_NAME>MARIA CATHARINA DE ALMEIDA SANTOS</PATIENT_NAME>
<PATIENT_BIRTHDATE>19391204</PATIENT_BIRTHDATE>
<PATIENT_SEX>MARIA CATHARINA DE ALMEIDA SANTOS</PATIENT_SEX>
<PATIENT_WEIGHT>67.8</PATIENT_WEIGHT>
<ACCESSION_NUMBER>22127077</ACCESSION_NUMBER>
<REQUESTING_PHYSICIAN>ERIDA THAYNA RIBEIRO</REQUESTING_PHYSICIAN>
<REQUESTING_PHYSICIAN_IDENTIFICATION>48002</REQUESTING_PHYSICIAN_IDENTIFICATION>
<REQUESTING_SERVICE>RADIOLOGY</REQUESTING_SERVICE>
<MODALITY>CR</MODALITY>
<REQUESTED_PROCEDURE_DESCRIPTION>RADIOGRAPHY OF THE CHEST (PA + SIDE + OBLIQUA)</REQUESTED_PROCEDURE_DESCRIPTION>
<DATE>20220513</DATE>
<TIME>113200</TIME>
</MWL_ITEM>


My database is structured like this

SELECT top 10 [na_accessionnumber]
      ,[co_patientid]
      ,[na_patientname]
      ,[na_patientsex]
      ,[na_patientbirthday]
      ,[na_patientweight]
      ,[na_patientheight]
      ,[co_performingid]
      ,[na_performingname]
      ,[na_performingcrm]
      ,[na_performinguf]
      ,[na_performingemail]
      ,[co_requesterid]
      ,[na_requestername]
      ,[na_requestercrm]
      ,[na_requesteruf]
      ,[na_requesteremail]
      ,[na_description]
      ,[na_bodypart]
      ,[na_studydate]
      ,[na_studytime]
      ,[na_modalityris]
      ,[na_password]
      ,[na_requireunit]
      ,[na_datetimerelease]
      ,[na_studyid]
      ,[on_machine]
      ,[na_unit]
      ,[na_customsearchfield]
      ,[na_datetimeintegrated]
      ,[co_worklistintegration]
      ,[na_requestingservice]
      ,[na_stepstatus]
      ,[na_operatorsname]
      ,[na_insuranceplan]
      ,[na_requestingdepartment]
      ,[na_requestedproceduredesc]
      ,[na_reqprocedurecodevalue]
      ,[na_reqprocedurecodschemedes]
      ,[na_patientrg]
      ,[na_patientcpf]
      ,[na_patientphone]
      ,[na_patientemail]
      ,[na_requesterphone]
      ,[na_clickvitawebprotocol]
      ,[na_clickvitawebpassword]
      ,[co_orderid]
  FROM worklistintegration

How could I do that in Mirth?

**Comment by rojao123** [comment]
up

**Reply by pacmano1** [reply]
I don't think you are getting a response because it doesn't seem like you have done some preliminary research on your own.   The question is also one that would covered in most training courses. 

Please read https://forums.mirthproject.io/articles/177082-forum-etiquette.    While that is the forums, the same guide applies here. 
 Note bullets 8, 9, and 10.     

This video on YouTube shows some database insert concepts: https://www.youtube.com/watch?v=KjSfXAXaCLY, the difference is your source side will be a DB reader and your data type is XML. The mapping concepts remain the same.
